ratio/ratio.go

```diff
@@ -38,10 +38,13 @@ func (r *Ratio) recordReplicatedPodSpec(replicas int32, spec corev1.PodSpec) *Ra
 	return r
 }
 
-// RecordPod collects all requests in the given Pod(s) and adds it to the ratio
+// RecordPod collects all requests in the given Pod(s), and adds it to the ratio
+// The function skips pods in phase `Succeeded` (e.g. completed job pods)
 func (r *Ratio) RecordPod(pods ...corev1.Pod) *Ratio {
 	for _, pod := range pods {
-		r.recordReplicatedPodSpec(1, pod.Spec)
+		if pod.Status.Phase != corev1.PodSucceeded {
```

Comment:
This probably erroneously includes resource requests of failed job pods, cf. https://pkg.go.dev/k8s.io/api/core/v1#PodPhase, however, if we ignore all pods in phase `Failed`, we'll erroneously skip pods in CrashLoopBackOff which should be counted.

Reply:
I generally agree with your reasoning that we should count anything other than succeeded pods, but AFAIK we're only billing running Pod. So IMO we should warn for the same thing as we bill.

Reply:
good point